app/src/main/kotlin/com/wire/android/media/audiomessage/ConversationAudioMessagePlayer.kt

```diff
@@ -413,14 +412,27 @@ class ConversationAudioMessagePlayer
         }
     }
 
+    private val getAssetMessageDeferredMap = ConcurrentHashMap<GetAssetMessageKey, Deferred<MessageAssetResult>>()
+
     private suspend fun getAssetMessage(
-        currentAccountResult: CurrentSessionResult,
+        userId: UserId,
         conversationId: ConversationId,
-        messageId: String
-    ) = coreLogic
-        .getSessionScope((currentAccountResult as CurrentSessionResult.Success).accountInfo.userId)
-        .messages
-        .getAssetMessage(conversationId, messageId)
+        messageId: String,
+    ): MessageAssetResult = withContext(Dispatchers.IO) {
```

Comment:
I think we should inject dispatchers into this MessagePlayer.
Otherwise it isn't fully testable.